mapper/orm/src/main/java/org/hibernate/search/mapper/orm/tenancy/spi/TenancyConfiguration.java

```diff
@@ -34,41 +41,122 @@ public class TenancyConfiguration {
 					.validate( value -> Contracts.assertNotNullNorEmpty( value, "value" ) )
 					.build();
 
-	public static TenancyConfiguration create(TenancyMode tenancyMode,
-			ConfigurationPropertySource configurationPropertySource) {
+	public static TenancyConfiguration create(TenancyMode tenancyMode, ConfigurationPropertySource configurationPropertySource,
+			BeanResolver beanResolver) {
 		String tenantIdsConfigurationPropertyKey = MULTI_TENANCY_TENANT_IDS.resolveOrRaw( configurationPropertySource );
 		switch ( tenancyMode ) {
 			case SINGLE_TENANCY:
-				return new TenancyConfiguration( Optional.of( Collections.emptySet() ),
-						tenantIdsConfigurationPropertyKey );
+				return new Single();
 			case MULTI_TENANCY:
-				return new TenancyConfiguration(
-						MULTI_TENANCY_TENANT_IDS.getAndMap( configurationPropertySource, LinkedHashSet::new ),
-						tenantIdsConfigurationPropertyKey );
+				// get from property:
```

Comment:
I guess this is a TODO?

I agree it would make sense to make this configurable, though I wonder if we couldn't also provide a decent default.

`toString()` is not the way to go, that's for sure, but assuming we can get some metadata from Hibernate ORM, we might be able to use something like `org.hibernate.type.descriptor.java.JavaType#toString(T)`/`org.hibernate.type.descriptor.java.JavaType#fromString(String)`...


I wonder if we actually need to make this configurable...

Reply:
we probably have to 😔 .... https://github.com/hibernate/hibernate-search/pull/3997#discussion_r1511047204